components/track/TrackShareDialog.vue

```diff
@@ -0,0 +1,93 @@
+<script lang="ts" setup>
+// import type { NuxtIconName } from '.nuxt/nuxt-icons';
+
+const { t } = useI18n();
+
+const props = defineProps<{
+  trackId: number;
+}>();
+
+const emit = defineEmits<{
+  close: [];
+}>();
+
+const message = ref<string | null>(null);
+
+// TODO: Make this dynamic
+const trackLink = `http://localhost:9001/track/${props.trackId}`;
+
+const setMessage = (newMessage: string) : Promise<null> => new Promise((resolve) => {
+    message.value = newMessage;
+    setTimeout(() => {
+      message.value = null;
+      resolve(null);
+    }, 2000);
+  });
+
+const shareButtons = {
+  facebook: {
+    text: t("track.dropdown.share-facebook"),
+    icon: "icon.facebook",
+    link: `https://www.facebook.com/sharer/sharer.php?u=${trackLink}`,
```

Comment:
the variable `trackLink` should be escaped using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent